monitoring/src/main/java/org/polypheny/db/monitoring/events/analyzer/DmlEventAnalyzer.java

```
@@ -0,0 +1,96 @@
+import org.polypheny.db.rel.RelNode;
+import org.polypheny.db.rel.RelRoot;
+
+
+@Slf4j
+public class DmlEventAnalyzer {
+
+    public static DmlDataPoint analyze( DmlEvent dmlEvent ) {
+        DmlDataPoint metric = DmlDataPoint
+                .builder()
+                .description( dmlEvent.getDescription() )
+                .Id( dmlEvent.getId() )
+                .fieldNames( dmlEvent.getFieldNames() )
+                .executionTime( dmlEvent.getExecutionTime() )
+                .rowCount( dmlEvent.getRowCount() )
+                .isSubQuery( dmlEvent.isSubQuery() )
+                .recordedTimestamp( dmlEvent.getRecordedTimestamp() )
+                .build();
+
+        RelRoot relRoot = dmlEvent.getRouted();
+        if ( relRoot != null ) {
+            RelNode node = relRoot.rel;
+            processRelNode( node, dmlEvent, metric );
+        }
+
+        if ( dmlEvent.isAnalyze() ) {
+            processDurationInfo( dmlEvent, metric );
+        }
+
+        return metric;
+    }
+
+
+    private static void processDurationInfo( DmlEvent dmlEvent, DmlDataPoint metric ) {
+        try {
+            InformationDuration duration = new Gson().fromJson( dmlEvent.getDurations(), InformationDuration.class );
+            getDurationInfo( metric, "Plan Caching", duration );
+            getDurationInfo( metric, "Index Lookup Rewrite", duration );
+            getDurationInfo( metric, "Constraint Enforcement", duration );
+            getDurationInfo( metric, "Implementation Caching", duration );
+            getDurationInfo( metric, "Index Update", duration );
+            getDurationInfo( metric, "Routing", duration );
+            getDurationInfo( metric, "Planning & Optimization", duration );
+            getDurationInfo( metric, "Implementation", duration );
+            getDurationInfo( metric, "Locking", duration );
+        } catch ( Exception e ) {
+            log.debug( "could not deserialize of get duration info" );
```

This hides an error...